<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1022</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Engine Parameters</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="conRod">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>20.375000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="steamPortWidth">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>0.655000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_21">
          <property name="text">
           <string>Valve Con Rod</string>
          </property>
         </widget>
        </item>
        <item row="1" column="5">
         <widget class="QDoubleSpinBox" name="bore">
          <property name="singleStep">
           <double>0.500000000000000</double>
          </property>
          <property name="value">
           <double>7.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QDoubleSpinBox" name="valveTopLand">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.270000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Valve Travel</string>
          </property>
          <property name="buddy">
           <cstring>valveTravel</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Valve Top Land</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Stroke</string>
          </property>
          <property name="buddy">
           <cstring>stroke</cstring>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Valve Width</string>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Bore</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Eccentric Advance</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Connecting Rod</string>
          </property>
          <property name="buddy">
           <cstring>conRod</cstring>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QDoubleSpinBox" name="exahustPortWidth">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QDoubleSpinBox" name="valveConRod">
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>22.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="5">
         <widget class="QDoubleSpinBox" name="eccentricAdvance">
          <property name="maximum">
           <double>360.000000000000000</double>
          </property>
          <property name="value">
           <double>120.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Valve Bottom Land</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QDoubleSpinBox" name="valveTravel">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>2.281000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Steam Port Width</string>
          </property>
          <property name="buddy">
           <cstring>steamPortWidth</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="stroke">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.500000000000000</double>
          </property>
          <property name="value">
           <double>8.500000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Exahust Port Width</string>
          </property>
          <property name="buddy">
           <cstring>exahustPortWidth</cstring>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QDoubleSpinBox" name="valveBottomLand">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.440000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="steamPortSpace">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>2.800000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>Steam Port Spacing</string>
          </property>
          <property name="buddy">
           <cstring>steamPortSpace</cstring>
          </property>
         </widget>
        </item>
        <item row="4" column="3">
         <widget class="QDoubleSpinBox" name="valveWidth">
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>4.660000000000000</double>
          </property>
         </widget>
        </item>
        <item row="5" column="0" colspan="6">
         <widget class="QCustomPlot" name="cyclePlot" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Animation</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="0">
         <widget class="myCustomPlot" name="valvePlot" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Position Control</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="4">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Bottom State:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QLabel" name="bottomCycle">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="8">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Crank -20</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="animationSlider">
           <property name="minimum">
            <number>-20</number>
           </property>
           <property name="maximum">
            <number>380</number>
           </property>
           <property name="value">
            <number>-20</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Crank 380</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="6">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Critical Point</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>Top State:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Current Crank Angle: </string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="7">
        <widget class="QComboBox" name="criticalPointSelect">
         <item>
          <property name="text">
           <string>Forward Intake</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Forward Cutoff</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Forward Release</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Return Compression</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Return Intake</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Return Cutoff</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Return Release</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Forward Compression</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="topCycle">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="currentAngle">
         <property name="minimum">
          <double>-20.000000000000000</double>
         </property>
         <property name="maximum">
          <double>380.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1022</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionUsage"/>
    <addaction name="actionReference"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionUsage">
   <property name="text">
    <string>Usage</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionReference">
   <property name="text">
    <string>Reference</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>myCustomPlot</class>
   <extends>QWidget</extends>
   <header>mycustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>stroke</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>conRod</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>368</x>
     <y>29</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>steamPortWidth</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>293</y>
    </hint>
    <hint type="destinationlabel">
     <x>508</x>
     <y>29</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>steamPortSpace</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>370</y>
    </hint>
    <hint type="destinationlabel">
     <x>644</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exahustPortWidth</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>447</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>23</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valveTravel</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>706</x>
     <y>191</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valveConRod</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>750</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valveTopLand</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>293</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>191</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valveBottomLand</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>370</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valveWidth</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>447</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>29</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>criticalPointSelect</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>drawValveDiagram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1000</x>
     <y>570</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>326</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>animationSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>sliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>593</y>
    </hint>
    <hint type="destinationlabel">
     <x>845</x>
     <y>434</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eccentricAdvance</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1001</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>819</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bore</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateEngineSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1001</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>844</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>criticalPointSelect</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setCriticalPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1000</x>
     <y>570</y>
    </hint>
    <hint type="destinationlabel">
     <x>818</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>currentAngle</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>currentAngleChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>222</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>957</x>
     <y>509</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>drawCycleDiagram()</slot>
  <slot>updateEngineSettings()</slot>
  <slot>drawValveDiagram()</slot>
  <slot>sliderChanged(int)</slot>
  <slot>setCriticalPoint(int)</slot>
  <slot>currentAngleChanged(double)</slot>
 </slots>
</ui>
